entry-generation/godot-entry-generator/src/main/kotlin/godot/entrygenerator/mapper/PropertyHintTypeMapper.kt

```diff
@@ -0,0 +1,298 @@
+            "godot.annotation.PropertyOfScript" ->
+                ClassName("godot.gdnative.godot_property_hint", "GODOT_PROPERTY_HINT_PROPERTY_OF_SCRIPT")
+            "godot.annotation.ObjectTooBig" ->
+                ClassName("godot.gdnative.godot_property_hint", "GODOT_PROPERTY_HINT_OBJECT_TOO_BIG")
+            "godot.annotation.NodePathValidTypes" ->
+                ClassName("godot.gdnative.godot_property_hint", "GODOT_PROPERTY_HINT_NODEPATH_VALID_TYPES")
+            "godot.annotation.SaveFile" ->
+                ClassName("godot.gdnative.godot_property_hint", "GODOT_PROPERTY_HINT_SAVE_FILE")
+            "godot.annotation.IntIsObjectId" ->
+                ClassName("godot.gdnative.godot_property_hint", "GODOT_PROPERTY_HINT_INT_IS_OBJECT_ID")
+            "godot.annotation.Max" -> ClassName("godot.gdnative.godot_property_hint", "GODOT_PROPERTY_HINT_MAX")
+            null -> ClassName("godot.gdnative.godot_property_hint", "GODOT_PROPERTY_HINT_NONE")
+            else -> throw IllegalStateException("Unknown annotation ${annotationDescriptor.fqName}")
+        }
+    }
+
+    fun mapAnnotationDescriptorToPropertyHintString(
+        propertyDescriptor: PropertyDescriptor,
+        annotationDescriptor: AnnotationDescriptor?
+    ): String {
+        if (!propertyDescriptor.annotations.getAnnotationValue(
+                REGISTER_PROPERTY_ANNOTATION,
+                REGISTER_PROPERTY_ANNOTATION_VISIBLE_IN_EDITOR_ARGUMENT,
+                true
+            ) && annotationDescriptor != null
+        ) {
+            throw IllegalStateException("You added the type hint annotation ${annotationDescriptor.fqName} to the property ${propertyDescriptor.name}. But the @RegisterProperty annotation is either not present or the isVisibleInEditor flag is not set to true")
+        }
+        return when (annotationDescriptor?.fqName?.asString()) {
+            "godot.annotation.IntRange" ->
+                getRangeTypeHint(annotationDescriptor, arrayOf(Int::class), propertyDescriptor)
+            "godot.annotation.FloatRange" ->
+                getRangeTypeHint(annotationDescriptor, arrayOf(Float::class), propertyDescriptor)
+            "godot.annotation.DoubleRange" ->
+                getRangeTypeHint(annotationDescriptor, arrayOf(Double::class), propertyDescriptor)
+            "godot.annotation.ExpRange" ->
+                getRangeTypeHint(annotationDescriptor, arrayOf(Float::class, Double::class), propertyDescriptor)
+            "godot.annotation.EnumTypeHint" -> getEnumTypeHint(propertyDescriptor)
+            "godot.annotation.ExpEasing" -> getExpEasingTypeHint(annotationDescriptor, propertyDescriptor)
+            "godot.annotation.Lenght" -> throw NotImplementedError("This annotation is not yet implemented") //getLengthTypeHint(annotationDescriptor, propertyDescriptor)
+            "godot.annotation.Flags" -> throw NotImplementedError("This annotation is not yet implemented")//getFlagsTypeHint(propertyDescriptor)
```

Comment:
Maybe we can add the name of the annotation in Exception message ?

Reply:
:+1: